File: tests/integration/aliases/python/step1/scenario_rename.py
# Copyright 2026, Pulumi Corporation.  All rights reserved.

from pulumi import Alias, ComponentResource, ResourceOptions


class Resource1(ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:Resource", name, None, opts)


# Scenario #1 - rename a resource
res1 = Resource1("res1")

File: tests/integration/aliases/go/step1/scenario_rename.go
// Copyright 2026, Pulumi Corporation.  All rights reserved.
//go:build !all
// +build !all

package main

import (
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

// renameComponent is a component resource
type renameComponent struct {
	pulumi.ResourceState
}

// Scenario #1 - rename a resource
func scenarioRename(ctx *pulumi.Context) error {
	fooComponent := &renameComponent{}
	return ctx.RegisterComponentResource("foo:component", "foo", fooComponent)
}
